dragonfly/implementations/roach2interface.py

```diff
@@ -0,0 +1,429 @@
+# -*- coding: utf-8 -*-
+"""
+"""
+
+
+from __future__ import absolute_import
+
+import logging
+import os
+import adc5g
+import numpy as np
+#import time
+from dripline import core
+from .ethernet_provider import EthernetProvider
+
+
+logger = logging.getLogger(__name__)
+
+#phasmid import
+
+try:
+    from .r2daq import ArtooDaq
+    logger.info('Imported ArtooDaq')
+
+except ImportError:
+    class ArtooDaq(object):
+        def __init__(self, *args, **kwargs):
+            raise RuntimeError("Dependency <ArtooDaq> not found but is required for ROACH2 support.")
+
+
+__all__ = []
+
+
+
+__all__.append('Roach2Provider')
+class Roach2Provider(ArtooDaq, core.Provider):
+    '''
+    A DAQProvider for interacting with the Roach
+    '''
+    def __init__(self, **kwargs):
+
+
+        core.Provider.__init__(self, **kwargs)
+
```

Comment:
no, only in the setup call after making a dictionary with the ports aso which is passed to ArtooDaq.__init__()